test/geometry/test_dsnt.py

```diff
@@ -0,0 +1,100 @@
+import pytest
+
+import kornia as kornia
+from test.common import dtype, device
+
+import torch
+from torch.testing import assert_allclose
+
+
+class TestRenderGaussian2d:
+    @pytest.fixture
+    def gaussian(self, device, dtype):
```

Comment:
just curious - did you actually test different dypes with cpu/gpu ? In my case I experienced strange behaviours with cpu and fp16

Reply:
Is there a convenient way to test all combinations of dtype and device? Normally I would parameterize the tests, but Kornia does not seem to be set up in that way.

Reply:
you have to do it manually by now:

 `pytest --typetest cpu/cpu --dtypetest float32/float64/float16`

I'm working to automatize this in CI, but in local not sure. Any suggestion ?

Reply:
In my projects I use parametrized fixtures, which will cause pytest to automatically run any test that uses the fixture multiple times with the different values. For example: https://github.com/anibali/tvl/blob/6078f671115c9cc73936b799caa98b2a0ca3fbae/tvl_backends/tests/conftest.py#L84-L86

Reply:
Upon further investigation it seems like CPU float16 tensors aren't really supported properly by PyTorch right now. Basic operations like addition, subtraction, multiplication, and division are throwing errors on my machine (eg `RuntimeError: "mul_cpu" not implemented for 'Half'`).

Reply:
Yeah, that's what more or less I saw. Float16 might require stuff like apex for testing (have you tried in tvl?). The way you propose can you exclude any of the option at any time from command line? I think that device and dtype are quite sensitive to the system environment. This is why I think it's better we can explicitly set it from user perspective.

Reply:
We could always use a mechanism similar to what you have now in combination with the parametrization. So you could have it set up so that you can run `pytest --device all --dtype all` or `pytest --device cpu --dtype float32,float64` or whatever you want.

Reply:
Sounds good

Reply:
@anibali I've been playing with fixture params and doesn't seem trivial (with minimal changes) to parametrize and make it dynamic based on the command line options. Pytest collects tests based on fixtures definition. Maybe there's a workaround for it.

Reply:
I'll put up a PR to demonstrate how it can be done.

Reply:
@edgarriba See https://github.com/kornia/kornia/pull/557